Fix bug #690: Write epoch string for SPICE with microsecond precision instead of 0.1 second

GorgiAstro opened the pull request:
* **Tickets addressed:** [#690](https://github.com/AVSLab/basilisk/issues/690)
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
The fix was described in [#690](https://github.com/AVSLab/basilisk/issues/690)

## Verification
A test case was added to `test_unitSpice.py`. Unfortunately, before SpiceInterface::initTimeData() is fixed, pytest will just stop the execution before the assert and not print any error. But after the fix, this test case runs successfully.

schaubh commented:
Thanks for contributing. I’ll take a look at this pool request as soon as possible.

schaubh reviewed (changes requested):
Thanks for the contribution.  Just some small items to address:

- the commit messages are too long and are getting truncated.  Please reduce the length of the first line and move some content to the following lines.  This is the case in both commits.
- add brief statement to release notes to `docs/source/Support/bskReleastNotes.rst`
- be sure to rebase on latest develop so we can merge right after final review

GorgiAstro commented:
> Thanks for the contribution. Just some small items to address:
> 
>     * the commit messages are too long and are getting truncated.  Please reduce the length of the first line and move some content to the following lines.  This is the case in both commits.
> 
>     * add brief statement to release notes to `docs/source/Support/bskReleastNotes.rst`
> 
>     * be sure to rebase on latest develop so we can merge right after final review

Thanks for the feedback, I addressed these 3 points, rebased onto latest `develop` and force-pushed.

schaubh reviewed (approved):
Thanks for the support!
